python/lsst/pipe/tasks/quickFrameMeasurement.py

```diff
@@ -0,0 +1,405 @@
+    def _calcMedianXxYy(objData):
+        """Return the median ixx and iyy for object in the image"""
+        medianXx = np.nanmedian([objData[i]['xx'] for i in objData.keys()])
+        medianYy = np.nanmedian([objData[i]['yy'] for i in objData.keys()])
+        return medianXx, medianYy
+
+    def _calcBrightestObjSrcNum(self, objData):
+        """Find the brightest source which passes the cuts among the sources."""
+        max70, max70srcNum = -1, -1
+        max25, max25srcNum = -1, -1
+
+        for srcNum in sorted(objData.keys()):  # srcNum not contiguous so don't use a list comp
+            # skip flag used rather than continue statements so we have all the
+            # metrics computed for debug purposes as this task is whack-a-mole
+            skip = False
+            xx = objData[srcNum]['xx']
+            yy = objData[srcNum]['yy']
+
+            xx = max(xx, 1e-9)  # need to protect against division by zero
+            yy = max(yy, 1e-9)  # because we don't `continue` on zero moments
+
+            if self.config.doExtendednessCut:
+                if xx > self.config.maxExtendedness or yy > self.config.maxExtendedness:
+                    skip = True
+
+            nonRoundness = xx/yy
+            nonRoundness = max(nonRoundness, 1/nonRoundness)
+            if nonRoundness > self.config.maxNonRoundness:
+                skip = True
+
+            if self.log.isDebugEnabled():
+                text = f"src {srcNum}: {objData[srcNum]['xCentroid']:.0f}, {objData[srcNum]['yCentroid']:.0f}"
+                text += f" - xx={xx:.1f}, yy={yy:.1f}, nonRound={nonRoundness:.1f}"
+                text += f" - ap70={objData[srcNum]['apFlux70']:,.0f}"
+                text += f" - ap25={objData[srcNum]['apFlux25']:,.0f}"
+                text += f" - skip={skip}"
+                self.log.debug(text)
+
+            if skip:
+                continue
+
+            ap70 = objData[srcNum]['apFlux70']
+            ap25 = objData[srcNum]['apFlux25']
+            if ap70 > max70:
+                max70 = ap70
+                max70srcNum = srcNum
+            if ap25 > max25:
+                max25 = ap25
+                max25srcNum = srcNum
+        if max70srcNum != max25srcNum:
+            self.log.warn("WARNING! Max apFlux70 for different object than with max apFlux25")
+
+        if max70srcNum >= 0:  # starts as -1, return None if nothing is acceptable
+            return max70srcNum
+        return None
+
+    def _measureFp(self, fp, exp):
+        """Run the measurements on a footprint."""
+        src = self.table.makeRecord()
+        src.setFootprint(fp)
+        self.centroider.measure(src, exp)
+        self.shaper.measure(src, exp)
+        self.apFluxer.measure(src, exp)
+        return src
+
+    @staticmethod
+    def _getDataFromSrcRecord(src):
+        """Extract the shapes and centroids from a source record."""
+        xx = np.sqrt(src['base_SdssShape_xx'])*2.355*.1  # 2.355 for FWHM, .1 for platescale
```

Comment:
Docstring - yes, for sure, that needs noting. For the platescale, how about instead of it being a config option, I pass in the exposure as well and get it with `exp.getWcs().getPixelScale().asArcseconds()`?

Reply:
That works as well.